### DISCUSSION TITLE
Run in Jupyter notebook

### DISCUSSION THREAD

**Discussion by r4dm** [rany2/edge-tts#139] [Q&A]
to run in a jupyter notebook, the code from the `basic_generation.py` needs to be corrected like this:

```
import nest_asyncio

nest_asyncio.apply()

async def amain() -> None:
    """Main function"""
    communicate = edge_tts.Communicate(TEXT, VOICE)
    await communicate.save(OUTPUT_FILE)

asyncio.run(amain())
```

Perhaps there is another solution?

**Comment by rany2** [comment]
This works for me in colab (maybe it works in jupyter as well), but their linter thinks I'm doing something wrong:

![image](https://github.com/rany2/edge-tts/assets/71077389/40d31b52-dbad-47c9-90a1-63feeb13bfa3)

**Comment by r4dm** [comment]
ty, works fine with no errors in vscode`s jupyter 

```
import edge_tts
from IPython.display import Audio
from IPython.display import display

VOICE = "en-GB-SoniaNeural"
OUTPUT_FILE = "test.mp3"

TEXT = "Hello World!"

communicate = edge_tts.Communicate(TEXT, VOICE)
await communicate.save(OUTPUT_FILE)

wn = Audio(OUTPUT_FILE, autoplay=True)
display(wn)
```
